table/sst_file_writer.cc

```diff
@@ -171,6 +186,23 @@ struct SstFileWriter::Rep {
       return Status::OK();
     }
 
+    assert(begin_key.size() >= ts_sz);
+    assert(end_key.size() >= ts_sz);
+    Slice begin_key_ts =
+        Slice(begin_key.data() + begin_key.size() - ts_sz, ts_sz);
+    [[maybe_unused]] Slice end_key_ts =
+        Slice(end_key.data() + end_key.size() - ts_sz, ts_sz);
+    assert(begin_key_ts.compare(end_key_ts) == 0);
+    if (strip_timestamp) {
+      // In this mode, we expect users to always provide a min timestamp.
+      if (internal_comparator.user_comparator()->CompareTimestamp(
+              begin_key_ts, MinU64Ts()) != 0) {
```

Comment:
I think it might make sense to validate `end_key_ts` as well just in case

Reply:
That makes sense! For range deletion, end key's timestamp slice should be exactly the same as the begin key's timestamp slice. I added an assertion for that, but it makes sense to check both of them for a possible InvalidArgument status too.